xdp-trafficgen fails to build against musl libc

on 1.4.0

```
xdp-trafficgen.c:793:33: error: passing argument 2 of 'getsockname' from incompatible pointer type [-Werror=incompatible-pointer-types]
  793 |         err = getsockname(sock, &local_saddr, &sockaddr_sz);
      |                                 ^~~~~~~~~~~~
      |                                 |
      |                                 struct sockaddr_in6 *
```

this is probably because local_saddr is sockaddr_in6 but getsockname takes struct sockaddr as documented. theoretically these are different pointers..

it's possible to get around this by casting it, but afaik that is still a strict aliasing violation (maybe?) so it would only be safe with -fno-strict-aliasing

--- a/xdp-trafficgen/xdp-trafficgen.c
+++ b/xdp-trafficgen/xdp-trafficgen.c
@@ -790,7 +790,7 @@ int do_tcp(const void *opt, __unused const char *pin_root_path)
 	}
 
 	sockaddr_sz = sizeof(local_saddr);
-	err = getsockname(sock, &local_saddr, &sockaddr_sz);
+	err = getsockname(sock, (struct sockaddr *)&local_saddr, &sockaddr_sz);
 	if (err) {
 		err = -errno;
 		pr_warn("Couldn't get local address: %s\n", strerror(-err));

